packages/blob/src/helpers.ts

```diff
@@ -32,12 +35,65 @@ export class BlobAccessError extends Error {
   }
 }
 
+export class BlobNotFoundError extends Error {
+  constructor() {
+    super('Vercel Blob: This store does not exist');
+  }
+}
+
+export class BlobSuspendedError extends Error {
+  constructor() {
+    super('Vercel Blob: This store has been suspended');
+  }
+}
+
 export class BlobUnknownError extends Error {
   constructor() {
     super('Vercel Blob: Unknown error, please visit https://vercel.com/help');
   }
 }
 
+type BlobApiErrorCodes =
+  | 'suspended_store'
+  | 'forbidden'
+  | 'not_found'
```

Comment:
store_not_found?

Reply:
that's the one defined from our vercel api and the worker

Reply:
From the Vercel API, when you try to authenticate a token, not_found without store_ prefix makes sense, because you're inquiring about an auth token related to a store. From the Vercel BLOB API, you can have either:
- store is not found
- or store is found but blob is not found

I believe they should have two different codes:
- not_found (blob, the resource you're targeting)
- store_not_found

i.e. on the DX side we need to provide details (am I trying to query a non-existing store, or is the blob just not here?)